src/corelibs/U2Gui/src/options_panel/OPWidgetFactory.h

```diff
@@ -93,11 +98,13 @@ class U2GUI_EXPORT OPFactoryFilterVisitor : public OPFactoryFilterVisitorInterfa
         return factoryAlphabetType == objectAlphabetType;
     }
     virtual bool atLeastOneAlphabetPass(DNAAlphabetType factoryAlphabetType);
+    bool atLeastOneDnaPass() const override;
 
 private:
     ObjectViewType objectViewType;
     DNAAlphabetType objectAlphabetType;
     QList<DNAAlphabetType> objectAlphabets;
+    QList<const DNAAlphabet *> alphabets;    // To check DNA.
```

Comment:
This field is not needed: it is a copy of 'objectAlphabets'
For whatever check you need you can use the original list.

Reply:
This is the thing I was talking about. It's not my changes, it's from @EvelinaBiserova. Of course I have checked them from my point of view, but I can't see everything and I am not sure how to fix it properly. And now you are asking me about stuff I haven't written at all. It's gotta be better to see PR's separately. I do not know how to resolve issues above, for now it's unneeded changes, so I just remove it and postpone this question up to the "PCR primer design" merge.